### PR TITLE
[PUB-3001/3002] Remove org injection in users

### PR DESCRIPTION
**Pull request by anafilipadealmeida**
## Description

- Delete no longer used userParser props and org rpc injection
- Delete INIT_USER on setCurrentOrganization in organizations middleware

## Context & Notes
https://buffer.atlassian.net/browse/PUB-3001
https://buffer.atlassian.net/browse/PUB-3002

## Checklist

-   [x] My code follows the code style and guidelines of this project. 
-   [ ] I have added tests to cover my changes.
-   [x] All new and existing tests passed.
-   [x] I have tested different user stories. 
-   [x] I have identified similar or related features and tested they are still working.
-   [x] I have performed a self-review of my own code.
-   [ ] I have tested my changes/additions in the latest Chrome, Firefox, and Safari.
-   [ ] I have commented my code, particularly in hard-to-understand areas.
-   [ ] I kept accessibility in mind by following the [a11y checklist.](https://www.notion.so/buffer/Workflow-Checklist-e64d86eb795140bcbfdc16d1c72e573f)
-   [ ] I have considered [security, abuse & compliance](https://www.notion.so/buffer/Engineering-Wiki-f34142d290304c35bebadf76cc9cc89e#cc6dcc7617184227b77da2e1b262a563) implications of my changes.

#### Staging Deployment

To visit the URL of the staging deployment, please click "Show all checks" at the bottom of this PR and click "details" next to `bufferbotbrains/cicd-buffer-publish-legacy`

### CODE REVIEW

File: packages/organizations/middleware.js [hunk 1]
```diff
@@ -95,11 +95,6 @@ export default ({ dispatch, getState }) => next => action => {
       break;
     }
 
-    // Temporarily dispatch user data, to inject new org data in it. To be removed after org_switcher rollout
-    case `setCurrentOrganization_${dataFetchActionTypes.FETCH_SUCCESS}`:
-      dispatch({ type: 'INIT_USER' });
-      break;
-
```
Comment (karelm) [inline packages/organizations/middleware.js:102]: :dancing_penguin: <<< wish that were available here

Its so great to see this comment come into reality 😂  🙌

**Review by karelm — APPROVED** [review]
Amazing work @anafilipadealmeida 🚀 :shipit:

**Review by hamstu — APPROVED** [review]
It's beautiful!

File: packages/server/parsers/src/orgParser.js [hunk 2]
```diff
@@ -59,6 +59,8 @@ module.exports = orgData => ({
     orgData.trial && orgData.trial.canStartBusinessTrial && orgData.isOwner,
   canStartProTrial:
     orgData.trial && orgData.trial.canStartProTrial && orgData.isOwner,
+  // add temporary way for buffer admin to connect personal ig profile
+  canSeeIGPersonalProfileConnect: orgData.isAdmin,
```
Comment (anafilipadealmeida) [inline packages/server/parsers/src/orgParser.js:63]: @albennett; I moved this to the org parser, as it was not accurate in the user 😅

**Pull request merged**